File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n  = nums.length;
        HashMap<Integer,Integer> dict = new HashMap<>();

        for(int i =0; i<n; i++){
            int rem = target - nums[i];
            if(dict.containsKey(rem)){
                return new int[]{dict.get(rem),i};

            }
            dict.put(nums[i],i);
        }
        return new int[]{};
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        dict ={}

        for i in range(n):
            rem = target -nums[i]
            if rem in dict:
                return [dict[rem],i]
            dict[nums[i]] = i
        